cidc_api/auth.py

```diff
@@ -38,11 +38,42 @@ def check_auth(
         TODO: role-based resource/method-level authorization
         """
         profile = self.token_auth(id_token)
+        is_authorized = self.role_auth(profile, allowed_roles, resource, method)
 
-        user = Users.create(profile["email"])
+        return is_authorized
 
+    def role_auth(
+        self, profile: dict, allowed_roles: List[str], resource: str, method: str
```

Comment:
shouldn't roles be like enum or something?

Reply:
It's true that it would be accurate to model roles with a type like `Role = Union[Literal['cidc-admin'], Literal['cimac-user'], ...]`, meaning that `allowed_roles: List[Role]` in the `role_auth` function signature, but I'm not sure we get anything in practice from adding this complexity to the type signature - especially since the `BearerAuth` class extends `eve.auth.TokenAuth`, whose function signature only guarantees that `check_auth` (and subsequently `role_auth`) will be passed `allowed_roles: List[str]`.

Reply:
I'm not sure I was able to follow your line of thought, I guess a list of strings will do.